src/Workspaces/Remote/ServiceHub/Services/LanguageServer.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.NavigateTo;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Newtonsoft.Json;
+using Roslyn.Utilities;
+using StreamJsonRpc;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.Remote
+{
+    // we need per language server for now since ILanguageClient
+    // doesn't allow multiple content types to be associated with
+    // one language server
+    internal class CSharpLanguageServer : LanguageServer
+    {
+        public CSharpLanguageServer(Stream stream, IServiceProvider serviceProvider)
+            : base(stream, serviceProvider, LanguageNames.CSharp)
+        {
+        }
+    }
```

Comment:
Do we actually need two subclasses? Since these do not override anything we could just create two instances of `LanguageServer` with different arguments.

Reply:
Never mind, I see now.

---
In reply to: [306995947](https://github.com/dotnet/roslyn/pull/37390#discussion_r306995947) [](ancestors = 306995947)